app/code/Magento/CatalogSearch/Model/Advanced.php

```diff
@@ -368,6 +368,13 @@ protected function getPreparedSearchCriteria($attribute, $value)
                 if (!empty($value['from']) || !empty($value['to'])) {
                     $from = '';
                     $to = '';
+                    if (is_array($value['from'])) {
+                        $value['from'] = $this->getFirstArrayElement($value['from']);
+                    }
+                    if (is_array($value['to'])) {
+                        $value['to'] = $this->getFirstArrayElement($value['to']);
+                    }
```

Comment:
I think, in case if we got an incorrect type - it's better to do assume that this parameter doesn't exist at all. I think we're expecting here the only string.

Reply:
@ihor-sviziev Hi! May be you are right and we should create a redirect if `$value['from']` or `$value['to']` is array. But I think we can check this arrays for some values and create a valid response.

cc: @andrewbess, @eduard13

Reply:
@SilinMykola I'm agree with @ihor-sviziev that we should ignore this parameter if it's not valid instead of trying to understand what we received and extract some data from it.
Looks like patch for all occasions.

Reply:
Ok, @Den4ik! Locally I removed this code and work on tests.

Reply:
The following comment is related to this discussion https://github.com/magento/magento2/issues/33589#issuecomment-893702664